src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AtomicMethods.kt

```diff
@@ -0,0 +1,288 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.util
+
+import java.util.concurrent.atomic.*
+import org.jetbrains.kotlinx.lincheck.util.AtomicMethodKind.*
+import org.jetbrains.kotlinx.lincheck.util.MemoryOrdering.*
+import java.lang.invoke.VarHandle
+
+internal data class AtomicMethodDescriptor(
+    val kind: AtomicMethodKind,
+    val ordering: MemoryOrdering,
+)
+
+internal enum class AtomicMethodKind {
```

Comment:
Do I understand correctly that you've added atomic method kinds for the upcoming model checker?

Reply:
Yes, for this too. 

When implementing this PR I went once again through the javadocs of all related APIs, so I decided to set-up the infrastructure to avoid doing this again in the future.

Reply:
Sounds reasonable 👍